### ISSUE TITLE
Rust panics with "no entry found for key" when documenting item with conflicting glob import

### ISSUE THREAD

**Issue by jonasbb** [rust-lang/rust#96981]
Running `cargo +stable doc` fails with
```
thread 'rustc' panicked at 'no entry found for key', src/librustdoc/passes/collect_intra_doc_links.rs:989:16
```
This is reproducible. The `pub use` is significant, a simple `use` does not trigger the error. The function name `read` is also significant. Naming it `copy` or `read_dir`, both functions existing in `tokio::fs` does *not* trigger the error.
This still fails with current nighly versions.

### Code

```Rust
pub use tokio::fs::*;

fn read() {}
```

```toml
tokio.default-features = false
tokio.version = "1.17"
tokio.features = ["fs"]
```

### Meta

`rustc --version --verbose`:
```
rustc 1.60.0 (7737e0b5c 2022-04-04)
binary: rustc
commit-hash: 7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c
commit-date: 2022-04-04
host: x86_64-unknown-linux-gnu
release: 1.60.0
LLVM version: 14.0.0
```

```
cargo 1.62.0-nightly (a44758ac8 2022-05-04)
release: 1.62.0-nightly
commit-hash: a44758ac805600edbb6ba51e7e6fb81a6077c0cd
commit-date: 2022-05-04
host: x86_64-unknown-linux-gnu
libgit2: 1.4.2 (sys:0.14.2 vendored)
libcurl: 7.80.0-DEV (sys:0.4.51+curl-7.80.0 vendored ssl:OpenSSL/1.1.1n)
os: Ubuntu 20.04 (focal) [64-bit]
```

### Error output

```
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.62.0-nightly (6dd68402c 2022-05-11) running on x86_64-unknown-linux-gnu

note: compiler flags: --crate-type lib

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: could not document `misc_utils`
```

<details><summary><strong>Backtrace stable</strong></summary>
<p>

```
thread 'rustc' panicked at 'no entry found for key', src/librustdoc/passes/collect_intra_doc_links.rs:916:16
stack backtrace:
   0:     0x7f560249786d - std::backtrace_rs::backtrace::libunwind::trace::hee598835bc88d35b
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f560249786d - std::backtrace_rs::backtrace::trace_unsynchronized::h9cdc730ba5cf5d72
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f560249786d - std::sys_common::backtrace::_print_fmt::h75aeaf7ed30e43fa
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x7f560249786d - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h606862f787600875
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x7f56024f74ec - core::fmt::write::he803f0f418caf762
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/fmt/mod.rs:1190:17
   5:     0x7f5602487a98 - std::io::Write::write_fmt::h70bc45872f37e7bb
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/io/mod.rs:1657:15
   6:     0x7f560249b847 - std::sys_common::backtrace::_print::h64d038cf8ac3e13e
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x7f560249b847 - std::sys_common::backtrace::print::h359300b4a7fccf65
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x7f560249b847 - std::panicking::default_hook::{{closure}}::hf51be35e2f510149
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:295:22
   9:     0x7f560249b510 - std::panicking::default_hook::h03ca0f22e1d2d25e
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:314:9
  10:     0x7f5602cb92c1 - rustc_driver[6eeb127b85fe99ac]::DEFAULT_HOOK::{closure#0}::{closure#0}
  11:     0x7f560249c124 - std::panicking::rust_panic_with_hook::h3b7380e99b825b63
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:702:17
  12:     0x7f560249bdd7 - std::panicking::begin_panic_handler::{{closure}}::h8e849d0710154ce0
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:588:13
  13:     0x7f5602497d34 - std::sys_common::backtrace::__rust_end_short_backtrace::hedcdaddbd4c46cc5
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:138:18
  14:     0x7f560249bae9 - rust_begin_unwind
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
  15:     0x7f5602464863 - core::panicking::panic_fmt::he1bbc7336d49a357
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
  16:     0x7f56024f42f1 - core::panicking::panic_display::hb184394d54dc8360
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:73:5
  17:     0x7f56024f429b - core::panicking::panic_str::ha42887cd972750c9
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:56:5
  18:     0x7f56024646d6 - core::option::expect_failed::h9159c9d16ed9fb7c
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/option.rs:1852:5
  19:     0x561d9e28b119 - rustdoc[a32c9c6374e2ed81]::passes::collect_intra_doc_links::resolve_associated_trait_item
  20:     0x561d9e2892e8 - <rustdoc[a32c9c6374e2ed81]::passes::collect_intra_doc_links::LinkCollector>::resolve_associated_item
  21:     0x561d9e287107 - <rustdoc[a32c9c6374e2ed81]::passes::collect_intra_doc_links::LinkCollector>::resolve
  22:     0x561d9e290eed - <rustdoc[a32c9c6374e2ed81]::passes::collect_intra_doc_links::LinkCollector>::resolve_link
  23:     0x561d9e28c5e9 - <rustdoc[a32c9c6374e2ed81]::passes::collect_intra_doc_links::LinkCollector as rustdoc[a32c9c6374e2ed81]::visit::DocVisitor>::visit_item
  24:     0x561d9e298f7a - <rustdoc[a32c9c6374e2ed81]::passes::collect_intra_doc_links::LinkCollector as rustdoc[a32c9c6374e2ed81]::visit::DocVisitor>::visit_inner_recur
  25:     0x561d9e28c9e4 - <rustdoc[a32c9c6374e2ed81]::passes::collect_intra_doc_links::LinkCollector as rustdoc[a32c9c6374e2ed81]::visit::DocVisitor>::visit_item
  26:     0x561d9e285a9c - rustdoc[a32c9c6374e2ed81]::passes::collect_intra_doc_links::collect_intra_doc_links
  27:     0x561d9e14e964 - <rustc_session[95a8f1c94d15e5dc]::session::Session>::time::<rustdoc[a32c9c6374e2ed81]::clean::types::Crate, rustdoc[a32c9c6374e2ed81]::core::run_global_ctxt::{closure#8}>
  28:     0x561d9e3387df - rustdoc[a32c9c6374e2ed81]::core::run_global_ctxt
  29:     0x561d9e14ede2 - <rustc_session[95a8f1c94d15e5dc]::session::Session>::time::<(rustdoc[a32c9c6374e2ed81]::clean::types::Crate, rustdoc[a32c9c6374e2ed81]::config::RenderOptions, rustdoc[a32c9c6374e2ed81]::formats::cache::Cache), rustdoc[a32c9c6374e2ed81]::main_options::{closure#0}::{closure#0}::{closure#1}::{closure#0}>
  30:     0x561d9e35f0c5 - <rustc_interface[afc566840a941718]::passes::QueryContext>::enter::<rustdoc[a32c9c6374e2ed81]::main_options::{closure#0}::{closure#0}::{closure#1}, core[2734bb1f7d9ec6a5]::result::Result<(), rustc_errors[bd800fb2a5e0b9f5]::ErrorReported>>
  31:     0x561d9e2ce090 - <rustc_interface[afc566840a941718]::interface::Compiler>::enter::<rustdoc[a32c9c6374e2ed81]::main_options::{closure#0}::{closure#0}, core[2734bb1f7d9ec6a5]::result::Result<(), rustc_errors[bd800fb2a5e0b9f5]::ErrorReported>>
  32:     0x561d9e32bf6c - rustc_span[4b4bb4534443c42f]::with_source_map::<core[2734bb1f7d9ec6a5]::result::Result<(), rustc_errors[bd800fb2a5e0b9f5]::ErrorReported>, rustc_interface[afc566840a941718]::interface::create_compiler_and_run<core[2734bb1f7d9ec6a5]::result::Result<(), rustc_errors[bd800fb2a5e0b9f5]::ErrorReported>, rustdoc[a32c9c6374e2ed81]::main_options::{closure#0}>::{closure#1}>
  33:     0x561d9e2bcd79 - rustc_interface[afc566840a941718]::interface::create_compiler_and_run::<core[2734bb1f7d9ec6a5]::result::Result<(), rustc_errors[bd800fb2a5e0b9f5]::ErrorReported>, rustdoc[a32c9c6374e2ed81]::main_options::{closure#0}>
  34:     0x561d9e0f8666 - rustdoc[a32c9c6374e2ed81]::main_options
  35:     0x561d9e299f5c - std[cd3cf6af28dff6de]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[afc566840a941718]::util::run_in_thread_pool_with_globals<rustdoc[a32c9c6374e2ed81]::main_args::{closure#0}, core[2734bb1f7d9ec6a5]::result::Result<(), rustc_errors[bd800fb2a5e0b9f5]::ErrorReported>>::{closure#0}, core[2734bb1f7d9ec6a5]::result::Result<(), rustc_errors[bd800fb2a5e0b9f5]::ErrorReported>>
  36:     0x561d9e14c2c9 - <<std[cd3cf6af28dff6de]::thread::Builder>::spawn_unchecked_<rustc_interface[afc566840a941718]::util::run_in_thread_pool_with_globals<rustdoc[a32c9c6374e2ed81]::main_args::{closure#0}, core[2734bb1f7d9ec6a5]::result::Result<(), rustc_errors[bd800fb2a5e0b9f5]::ErrorReported>>::{closure#0}, core[2734bb1f7d9ec6a5]::result::Result<(), rustc_errors[bd800fb2a5e0b9f5]::ErrorReported>>::{closure#1} as core[2734bb1f7d9ec6a5]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  37:     0x7f56024a7d03 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hf70ac038171e3e1a
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/alloc/src/boxed.rs:1853:9
  38:     0x7f56024a7d03 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::he6690128792365ad
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/alloc/src/boxed.rs:1853:9
  39:     0x7f56024a7d03 - std::sys::unix::thread::Thread::new::thread_start::ha07928d93d5a5ec9
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys/unix/thread.rs:108:17
  40:     0x7f5602289e5d - start_thread
  41:     0x7f560230f8f0 - __clone3
  42:                0x0 - <unknown>
```

</p>
</details>

<details><summary><strong>Backtrace nightly</strong></summary>
<p>

```
thread 'rustc' panicked at 'no entry found for key', src/librustdoc/passes/collect_intra_doc_links.rs:989:16
stack backtrace:
   0:     0x7efd987bce5d - std::backtrace_rs::backtrace::libunwind::trace::h7a9c384780f2ae7b
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7efd987bce5d - std::backtrace_rs::backtrace::trace_unsynchronized::heb75d166ca8b8236
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7efd987bce5d - std::sys_common::backtrace::_print_fmt::h322cd7edf00cb3b5
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x7efd987bce5d - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2bdd168863171e89
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x7efd98818adc - core::fmt::write::he97f80a3d639b1e1
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/core/src/fmt/mod.rs:1196:17
   5:     0x7efd987ae5e1 - std::io::Write::write_fmt::hfad334cf5c2887e6
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/io/mod.rs:1654:15
   6:     0x7efd987bfb75 - std::sys_common::backtrace::_print::hb3fd4d16502c8b4b
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x7efd987bfb75 - std::sys_common::backtrace::print::h20c8699087b98c21
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x7efd987bfb75 - std::panicking::default_hook::{{closure}}::hceced8f030c627d7
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/panicking.rs:295:22
   9:     0x7efd987bf7e9 - std::panicking::default_hook::hf622cc58f7fb1da8
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/panicking.rs:314:9
  10:     0x7efd98f834a1 - rustc_driver[8eec3efc1f3aef49]::DEFAULT_HOOK::{closure#0}::{closure#0}
  11:     0x7efd987c0346 - std::panicking::rust_panic_with_hook::h6a32c5574ceba44d
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/panicking.rs:702:17
  12:     0x7efd987c0147 - std::panicking::begin_panic_handler::{{closure}}::h33a7eb85cc49019a
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/panicking.rs:588:13
  13:     0x7efd987bd314 - std::sys_common::backtrace::__rust_end_short_backtrace::hbc4249165d4ceafc
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/sys_common/backtrace.rs:138:18
  14:     0x7efd987bfe79 - rust_begin_unwind
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/panicking.rs:584:5
  15:     0x7efd98785133 - core::panicking::panic_fmt::hea2f72507c137b32
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/core/src/panicking.rs:142:14
  16:     0x7efd98815741 - core::panicking::panic_display::h829100d766eff741
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/core/src/panicking.rs:72:5
  17:     0x7efd988156eb - core::panicking::panic_str::hce7fb8f66b788a84
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/core/src/panicking.rs:56:5
  18:     0x7efd98784fa6 - core::option::expect_failed::haa2621cfe315379c
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/core/src/option.rs:1855:5
  19:     0x55d87e128d49 - rustdoc[11fd5c4f876ad31e]::passes::collect_intra_doc_links::resolve_associated_trait_item
  20:     0x55d87e1274ee - <rustdoc[11fd5c4f876ad31e]::passes::collect_intra_doc_links::LinkCollector>::resolve_associated_item
  21:     0x55d87e1246e3 - <rustdoc[11fd5c4f876ad31e]::passes::collect_intra_doc_links::LinkCollector>::resolve
  22:     0x55d87e12a871 - <rustdoc[11fd5c4f876ad31e]::passes::collect_intra_doc_links::LinkCollector as rustdoc[11fd5c4f876ad31e]::visit::DocVisitor>::visit_item
  23:     0x55d87e13d09a - <rustdoc[11fd5c4f876ad31e]::passes::collect_intra_doc_links::LinkCollector as rustdoc[11fd5c4f876ad31e]::visit::DocVisitor>::visit_inner_recur
  24:     0x55d87e12e258 - <rustdoc[11fd5c4f876ad31e]::passes::collect_intra_doc_links::LinkCollector as rustdoc[11fd5c4f876ad31e]::visit::DocVisitor>::visit_item
  25:     0x55d87e1229dc - rustdoc[11fd5c4f876ad31e]::passes::collect_intra_doc_links::collect_intra_doc_links
  26:     0x55d87e1dbad7 - rustdoc[11fd5c4f876ad31e]::main_options::{closure#0}::{closure#0}::{closure#1}
  27:     0x55d87e177659 - <rustc_interface[d03aac2056e219e8]::passes::QueryContext>::enter::<rustdoc[11fd5c4f876ad31e]::main_options::{closure#0}::{closure#0}::{closure#1}, core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>>
  28:     0x55d87e0c418f - <rustc_interface[d03aac2056e219e8]::interface::Compiler>::enter::<rustdoc[11fd5c4f876ad31e]::main_options::{closure#0}::{closure#0}, core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>>
  29:     0x55d87e228ab6 - rustc_span[5dae44024ba07c23]::with_source_map::<core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>, rustc_interface[d03aac2056e219e8]::interface::create_compiler_and_run<core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>, rustdoc[11fd5c4f876ad31e]::main_options::{closure#0}>::{closure#1}>
  30:     0x55d87e0ec654 - rustc_interface[d03aac2056e219e8]::interface::create_compiler_and_run::<core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>, rustdoc[11fd5c4f876ad31e]::main_options::{closure#0}>
  31:     0x55d87df41b45 - <scoped_tls[9a81942410e5ea2f]::ScopedKey<rustc_span[5dae44024ba07c23]::SessionGlobals>>::set::<rustdoc[11fd5c4f876ad31e]::main_args::{closure#0}, core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>>
  32:     0x55d87e0ece2f - std[87b888c30fb8b53d]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[d03aac2056e219e8]::util::run_in_thread_pool_with_globals<rustdoc[11fd5c4f876ad31e]::main_args::{closure#0}, core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>>::{closure#0}, core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>>
  33:     0x55d87e18e159 - <<std[87b888c30fb8b53d]::thread::Builder>::spawn_unchecked_<rustc_interface[d03aac2056e219e8]::util::run_in_thread_pool_with_globals<rustdoc[11fd5c4f876ad31e]::main_args::{closure#0}, core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>>::{closure#0}, core[c5a39be06b715266]::result::Result<(), rustc_errors[ddf493a40e6cbb28]::ErrorGuaranteed>>::{closure#1} as core[c5a39be06b715266]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  34:     0x7efd987ca253 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h23107667314c5666
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/alloc/src/boxed.rs:1872:9
  35:     0x7efd987ca253 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h8a72a46bb071903a
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/alloc/src/boxed.rs:1872:9
  36:     0x7efd987ca253 - std::sys::unix::thread::Thread::new::thread_start::h81657d69ed68fad8
                               at /rustc/6dd68402c5d7da168f87d8551dd9aed1d8a21893/library/std/src/sys/unix/thread.rs:108:17
  37:     0x7efd986fa609 - start_thread
                               at /build/glibc-sMfBJT/glibc-2.31/nptl/pthread_create.c:477:8
  38:     0x7efd984a3163 - clone
  39:                0x0 - <unknown>
```

</p>
</details>

**Comment by jonasbb** [comment]
I could produce this error also on stable 1.60. I updated the original post to include the description and backtrace for stable.

**Comment by petrochenkov** [comment]
This issue was fixed in Rust 1.63.
There's no minimized reproduction to add as a test, so I'll just close this issue.
